private/pkg/oauth2/client.go

```diff
@@ -28,6 +29,13 @@ import (
 	"go.uber.org/multierr"
 )
 
+var (
+	// ErrUnsupported is returned when we receive an unsupported response from the server.
+	//
+	// TODO(go1.23): replace by errors.ErrUnsupported once it is available.
```

Comment:
I thought this was added in go1.21?

> The new [ErrUnsupported](https://go.dev/pkg/errors/#ErrUnsupported) error provides a standardized way to indicate that a requested operation may not be performed because it is unsupported.

https://go.dev/doc/go1.21#errorspkgerrors

Reply:
Ops yep, depends on https://github.com/bufbuild/buf/pull/3123